homeassistant/components/swiss_public_transport/__init__.py

```diff
@@ -131,9 +160,9 @@ async def async_migrate_entry(
             config_entry, unique_id=new_unique_id, minor_version=2
         )
 
-    if config_entry.version < 2:
-        # Via stations now available, which are not backwards compatible if used, changes unique id
-        hass.config_entries.async_update_entry(config_entry, version=2, minor_version=1)
+    if config_entry.version < 3:
+        # Via stations and time/offset settings now available, which are not backwards compatible if used, changes unique id
+        hass.config_entries.async_update_entry(config_entry, version=3, minor_version=1)
```

Comment:
This needs to update the unique id again (like when you go from version 1 to 2).
Could also consider to remove the unique id instead...

Reply:
No it does not, the unique_id generation function is backwards compatible. When you use the `now` option, which is the only one available in version 2, the unique_id generated is the same, which is why no migration is needed.

Reply:
Unique id can be removed in a follow up PR, as it does not really provide a benefit.